src/Illuminate/Support/Password.php

```diff
@@ -0,0 +1,241 @@
+    {
+        $this->withNumbers = $withNumbers;
+
+        return $this;
+    }
+
+    /**
+     * Setting the flag for using special symbols.
+     *
+     * @param  bool  $withSymbols
+     * @return $this
+     */
+    public function withSymbols(bool $withSymbols): self
+    {
+        $this->withSymbols = $withSymbols;
+
+        return $this;
+    }
+
+    /**
+     * @param  bool  $withSpaces
+     * @return $this
+     */
+    public function withSpaces(bool $withSpaces): self
+    {
+        $this->withSpaces = $withSpaces;
+
+        return $this;
+    }
+
+    /**
+     * Setting valid letters.
+     *
+     * @param  string[]  $letters
+     * @return $this
+     */
+    public function allowedLetters(array $letters): self
+    {
+        $this->allowedLetters = $letters;
+
+        return $this;
+    }
+
+    /**
+     * Setting valid numbers.
+     *
+     * @param  string[]  $numbers
+     * @return $this
+     */
+    public function allowedNumbers(array $numbers): self
+    {
+        $this->allowedNumbers = $numbers;
+
+        return $this;
+    }
+
+    /**
+     * Setting valid special symbols.
+     *
+     * @param  string[]  $symbols
+     * @return $this
+     */
+    public function allowedSymbols(array $symbols): self
+    {
+        $this->allowedSymbols = $symbols;
+
+        return $this;
+    }
+
+    /**
+     * Generate password.
+     *
+     * @return string
+     *
+     * @throws Exception
+     */
+    public function build(): string
+    {
+        $minLength = (int) $this->withLetters +
+            (int) $this->withNumbers +
+            (int) $this->withSymbols +
+            (int) $this->withSpaces;
+
+        if ($this->length < $minLength) {
+            throw new PasswordInvalidLengthArgument(
+                sprintf('Minimum password length must be %d characters', $minLength)
+            );
+        }
+
+        $letters = $this->allowedLetters;
+        $numbers = $this->allowedNumbers;
+        $symbols = $this->allowedSymbols;
+
+        $combination = (new Collection)
+            ->when($this->withLetters, fn ($c) => $c->merge($letters))
+            ->when($this->withNumbers, fn ($c) => $c->merge($numbers))
+            ->when($this->withSymbols, fn ($c) => $c->merge($symbols));
+
+        if ($combination->count() === 0) {
+            throw new PasswordInvalidArgumentsException('Not a valid set of parameters for generating a password');
+        }
+
+        $chars = new Collection();
+        while ($chars->count() < $this->length) {
+            $chars = $chars->merge($combination);
+        }
+
+        return $chars
+            ->shuffle()
+            ->take($this->length)
+            ->when($this->withLetters, fn ($c) => $c->splice(1)->merge($letters[array_rand($letters)]))
+            ->when($this->withNumbers, fn ($c) => $c->splice(1)->merge($numbers[array_rand($numbers)]))
+            ->when($this->withSymbols, fn ($c) => $c->splice(1)->merge($symbols[array_rand($symbols)]))
```

Comment:
I know this has been closed already, but it's worth pointing out that `array_rand()` isn't cryptographically secure and shouldn't be used for password generation.